mypy/newsemanal/semanal.py

```diff
@@ -2563,6 +2563,9 @@ def analyze_member_lvalue(self, lval: MemberExpr, explicit_type: bool, is_final:
             if cur_node and is_final:
                 # Overrides will be checked in type checker.
                 self.fail("Cannot redefine an existing name as final", lval)
+            if (not lval.node and cur_node and isinstance(cur_node.node, Var) and
+                    cur_node.node.is_inferred and explicit_type):
```

Comment:
TBH I don't follow the logic here, maybe add a comment explaining this?

Reply:
Yes, I can write a comment. But before, let me explain: Fail if passing first time, the node exists and it's a `Var` node, it's type is inferred, and new assignment has an explicit type.

Reply:
Basically yes. But please try to avoid this:
```
x += 1  # Increment x
```
Describe the intent and the code achieves it.

Reply:
@onlined Are you going to add this comment (also in the old analyzer)?

Reply:
Yes, I will.